src/mapping/star_align/script.py

```diff
@@ -0,0 +1,161 @@
+import os
+import re
+import tempfile
+import subprocess
+from pathlib import Path
+import tarfile
+import gzip
+import shutil
+
+## VIASH START
+par = {
+  'genomeDir': 'resources_test/cellranger_tiny_fastq/cellranger_tiny_ref/star/',
+  'readFilesIn': [
+    'resources_test/cellranger_tiny_fastq/cellranger_tiny_fastq/tinygex_S1_L001_R1_001.fastq.gz',
+    'resources_test/cellranger_tiny_fastq/cellranger_tiny_fastq/tinygex_S1_L001_R2_001.fastq.gz',
+    'resources_test/cellranger_tiny_fastq/cellranger_tiny_fastq/tinygex_S1_L002_R1_001.fastq.gz',
+    'resources_test/cellranger_tiny_fastq/cellranger_tiny_fastq/tinygex_S1_L002_R2_001.fastq.gz'
+  ]
+}
+meta = {
+  'config': 'src/mapping/star_align/.config.vsh.yaml',
+  'cpus': None,
+  'memory_b': None,
+  'memory_kb': None,
+  'memory_mb': None,
+  'memory_gb': None,
+  'memory_tb': None,
+  'memory_pb': None,
+  'temp_dir': '/tmp'
+}
+## VIASH END
+
+# fastqgz_regex = r'([A-Za-z0-9\-_\.]+)_S(\d+)_L(\d+)_([RI]\d+)_(\d+)\.fastq(\.gz)?'
+fastqgz_regex = r'([A-Za-z0-9\-_\.]+)_S(\d+)_L(\d+)_(R\d+)_(\d+)\.fastq(\.gz)?'
+
+def is_gz_file(filepath):
+  with open(filepath, 'rb') as test_f:
+    return test_f.read(2) == b'\x1f\x8b'
+
+# rename keys
+to_rename = {'input': 'readFilesIn', 'reference': 'genomeDir', 'output': 'outFileNamePrefix'}
+par = { to_rename[key] if key in to_rename.keys() else key: value
+  for key, value in par.items() }
+
+# create output dir if need be
+if not os.path.exists(par["outFileNamePrefix"]):
+  os.mkdir(par["outFileNamePrefix"])
+
+# make sure there is a trailing /
+par["outFileNamePrefix"] = par["outFileNamePrefix"] + "/"
+
+with tempfile.TemporaryDirectory(prefix="star-", dir=meta["temp_dir"]) as temp_dir:
+  print(">> Check whether input files are directories", flush=True)
+  new_read_files_in = []
+  for value in par["readFilesIn"]:
+    value_path = Path(value)
+    if value_path.is_dir():
+      print(f"Input '{value_path}' is a directory, traversing to see if we can detect any FASTQ files.", flush=True)
```

Comment:
If flush is needed, better use the `logging` module

Reply:
Kekko desu!

I prefer having to specify `, flush=True` instead of having 10 extra lines of code at the top :)